llvm/lib/Transforms/Scalar/EarlyCSE.cpp

```diff
@@ -1525,6 +1525,11 @@ bool EarlyCSE::processNode(DomTreeNode *Node) {
       }
     }
 
+    // Make sure stores prior to a potential unwind are not removed, as the
+    // caller may read the memory.
+    if (Inst.mayThrow())
```

Comment:
All the target memory store intrinsics have `nounwind`, right?

Reply:
I think so. It's the default for intrinsics.